src/app/contact-form/contact-form.component.ts

```diff
@@ -32,21 +35,25 @@ export class ContactFormComponent implements OnInit {
     this.hide.emit();
   }
 
-  send(): void {
+  send(contactForm): void {
     this.backendService.post(
       this.requestData,
       `/device-availability/${this.availability.id}/contact`,
       { headers: new HttpHeaders({ 'Content-Type': 'application/json' }) })
       .subscribe(
         () => {
-          this.requestData = {
-            name: '',
-            email: '',
-            message: '',
-          };
+          contactForm.form.reset();
           this.msgSent.emit();
         },
-        (err) => this.emailError = err.error.message,
+        (err) => {
+          this.emailError = err.error.message;
+          this.requestData.captcha = '';
+        },
       );
   }
+
+  // Captcha error
+  onError(error: any) {
+    this.emailError = 'Error while validating captcha.';
```

Comment:
1. I think that this is a good time to rename the variable as it does not only store error messages concerning the email address.

2. Rename the function to onCaptchaError?

Reply:
done